src/pcre2_util.h

```diff
@@ -44,26 +44,52 @@ POSSIBILITY OF SUCH DAMAGE.
 /* Assertion macros */
 
 #ifdef HAVE_BUILTIN_UNREACHABLE
-#define PCRE2_UNREACHABLE() __builtin_unreachable()
-#elif HAVE_BUILTIN_ASSUME
-#define PCRE2_UNREACHABLE() __assume(0)
+#define PCRE2_DEADTRAP() __builtin_unreachable()
+#elif defined(HAVE_BUILTIN_ASSUME)
+#define PCRE2_DEADTRAP() __assume(0)
 #else
-#define PCRE2_UNREACHABLE() do {} while(0)
+#warning "missing implementation of PCRE2_UNREACHABLE()"
```

Comment:
This looks like a strong requirement

Reply:
I know and I didn't like it either, but how else can we let people know that their compiler might silently continue running code after reaching something labeled as `PCRE2_UNREACHABLE()`.

This whole funny business with #489 where we could reasonable think it was ok to delete the fail safe code twice in a month, makes me think this is too confusing and needs at least a stronger "heads up".

Reply:
well ok

Reply:
BTW removed it from the latest revisions, because it was too painful to have it when building on anything outside the 3 compilers we have implementations for, but that means we also have to be extra careful in the opposite direction, because something like:

```c
  case THIS:
     do_good();
     PCRE2_UNREACHABLE();
  case THAT:
     do_bad();
```

is perfectly valid code and will build cleanly on those 3 compilers we have in CI, but result in a dangerous fall through when built by a different compiler.

Reply:
Carlo, just a thought... Aside from gcc, clang, and MS Visual C++, do you know any other C compilers which are commonly used to build PCRE2? If so, I can do a bit of research and see if they have a similar built-in.

I understand that even so, the issue will still remain that *some* users, *somewhere* may use a compiler for which PCRE2_UNREACHABLE is not implemented. But it wouldn't be bad to have implementations for as many compilers as possible.

Reply:
> do you know any other C compilers which are commonly used to build PCRE2?

Oracle's (AKA Sun) Studio, the latest release doesn't have this (with the C++ version implemented as a NOP internally, so it will even be detected and used but still behave as erratically as described above). IBM's  xlc and DEC's for the VMS are also likely common.

The important point though is not that we "could" add support for them, but the fact that we are opening ourselves to issues like this by using this builtins() that are specifically described as triggering "undefined behaviour" and so will need to be more careful going forward on our code review (I guess) or have to invent something else to protect us from that.